base/int.jl

```diff
@@ -122,7 +123,8 @@ julia> iseven(10)
 true
 ```
 """
-iseven(n::Integer) = !isodd(n)
+iseven(n::Number) = isreal(n) && iseven(real(n))
+iseven(n::Real) = isinteger(n) && iszero(rem(Integer(n), 2))
```

Comment:
Doesn't the cast to Integer here defeat the purpose?
```
julia> Integer(2.0^65)
ERROR: InexactError: Int64(3.6893488147419103e19)
Stacktrace:
 [1] Int64
   @ ./float.jl:747 [inlined]
 [2] Integer(x::Float64)
   @ Core ./boot.jl:789
 [3] top-level scope
   @ REPL[2]:1
```

Reply:
No, because there is a specialized method for `AbstractFloat`.

This is just the fallback definition, which works for things like `Complex{Int}` and `Rational{Int}`.

Reply:
Ah, right, thanks